src/operator/controllers/intents_controller.go

```diff
@@ -133,26 +148,25 @@ func (r *IntentsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 }
 
 func (r *IntentsReconciler) intentsReconcilerInit(ctx context.Context) error {
-	if !telemetrysender.IsTelemetryEnabled() {
-		// When telemetry is disabled no logic should run by the telemetry reconciler. In case that a previous run of
-		// the operator had telemetry enabled we must remove the telemetry reconciler finalizers from all the CRDs.
-		err := r.RemoveFinalizerFromAllResources(ctx, otterizev1alpha2.OtterizeTelemetryReconcilerFinalizerName)
-		if err != nil {
-			return err
-		}
+	err := r.RemoveLegacyFinalizer(ctx)
```

Comment:
RemoveLegacyFinalizer happens here and removes all finalizers for all resources, but also in the reconcilergroup per-resource (which is the correct behavior)